### ISSUE TITLE
How to access transformer's prediction at each timestep, during decoding

### ISSUE THREAD

**Issue by reubenharry** [tensorflow/tensor2tensor#887]
### Description

(Not an issue, so much as a question)

For my purposes, it would be extremely useful to be able to access the transformer's prediction at each timestep of decoding (e.g. in greedy search, e.g. for a trained English to German translation model). I've tinkered with the code a bit and can maybe hack something together, but wondered if there was a direct way.

Ideally, I'd like to be able to see the distribution over tokens at each timestep, in a greedy decoding, and later down the road, make a model that just returns the next logits (i.e. like the symbols_to_logits_fn function).

Thanks so much - t2t is awesome and is working super well!

**Comment by mabergerx** [comment]
+1, any ideas for this? Thanks!
